ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+    
+    While being quite different from Orleans, Akka is also an implementation of the spirit embodied by the Reactive Manifesto.
+
+Besides these differences in the guiding principles of both tools one particular difference deserves further elaboration. The modeling of Actors via strongly typed interfaces and proxy objects is the most prominent difference between Orleans and Akka; the latter includes the same functionality via the `akka.actor.TypedActor` feature, but that has been deprecated for a long time, awaiting its replacement by an equally type-safe but explicitly message-oriented abstraction. We have observed that offering classical method-based interaction between objects takes away many of the advantages of the purely message-driven nature of the Actor Model, in particular it is impossible to work with the underlying stream of messages for routing, aggregation, transformation and so on because this essential nature of the interaction is hidden from the programming model, it is abstracted away. Another manifestation of this difference is that TypedActor and Orleans Grains have as many entry points into the behavior of an object (Actor) as there are exposed methods while in normal Actors there is just one behavior that receives the next message at any given time. This leads to the loss of dynamic behavior changes (Grains cannot offer the equivalent of Akka’s `context.become()` and TypedActor has the same restriction). This also shines through in the rough edges around reentrancy for in-Grain continuations using async–await: while a classical Actor can—and must—freely choose which requests to ignore or buffer and which to answer during the period where another Actor’s response is awaited, Grains can only work in either the fully blocked or fully reentrant modes, limiting the user’s choices to a safe one and a fast one.
```

Comment:
Thanks for the background on TPL, I was not aware of some of the aspects and will clarify the text once I’m clear on the implications.